lib/pbench/server/api/resources/query_apis/datasets_detail.py

```diff
@@ -0,0 +1,266 @@
+        except KeyError:
+            keys = [k for k in ("user", "name", "start", "end") if k not in json_data]
+            self.logger.info("Missing required JSON keys {}", ",".join(keys))
+            abort(400, message=f"Missing request data: {','.join(keys)}")
+
+        # We need to support a query for public data without requiring
+        # authorization; however if an Authorization header is given,
+        # it must be of recommended JWT "Bearer" schema, and have a
+        # non-empty token.
+        #
+        # TODO: validate the token with the user management object.
+        authorization = request.headers.get("Authorization")
+        session_token = None
+        if authorization:
+            type, token = authorization.split()
+            if type.lower() == "bearer":
+                session_token = token
+
+            if not session_token:
+                self.logger.warn(
+                    '"Authorization" header specifies unsupported or missing '
+                    ' schema; use "Authorization: Bearer <JWT-token>"'
+                )
+                abort(401, message="invalid user authorization")
+
+        try:
+            start = parser.parse(start_arg).replace(day=1)
+            end = parser.parse(end_arg).replace(day=1)
+        except Exception as e:
+            self.logger.info(
+                "Invalid start or end time string: {}, {}: {}", start_arg, end_arg, e
+            )
+            abort(400, message="Invalid start or end time string")
+
+        # We have nothing to authorize against yet, but print the specified
+        # user and authorization token to prove we got them.
+        self.logger.info(
+            "Return dataset {} for user {}, prefix {}, authorization {}: ({} - {})",
+            name,
+            user,
+            self.prefix,
+            session_token,
+            start,
+            end,
+        )
+
+        payload = {
+            "query": {
+                "bool": {
+                    "filter": [
+                        {"match": {"run.name": name}},
+                        {"match": get_user_term(user)},
+                    ]
+                }
+            },
+            "sort": "_index",
+        }
+
+        # TODO: the big assumption here involves the index version we're
+        # referencing. While it's possible the root index name could change
+        # at some point, advancing the version is going to happen more
+        # often.
+        #
+        # So ... do we need to change the code here each time? The old
+        # dashboard config is hardcoded for each version, so one option is
+        # to add a version for each index to the pbench-server.cfg file.
+        #
+        # When we have a full live Python server, persisting the template
+        # information in Redis or postgreSQL probably makes sense; maybe
+        # we just hardcode until then?
+        uri_fragment = gen_month_range(self.prefix, ".v6.run-data.", start, end)
```

Comment:
Are we now telling the dashboard these strings via the config APIs?  Seems like we don't want to hard code this.